### PR TITLE
Added ability to specify image transformations

### LINKED ISSUE #36: Feature - Image types should allow transformations
## Overview
Salsify supports transforming images, this module should allow developers to take advantage of this.

## Proposed Solution
Add image type configuration options

## Additional context
https://getstarted.salsify.com/help/transforming-image-files

### PR DESCRIPTION
**Pull request by mak001**
- relies on salsify to transform the images

Resolves #36

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -284,21 +284,9 @@ Dynamic\Salsify\Model\Mapper.example:
...
- - `Thumbnail` or `Thumb`
- - `Pad`
- - `Fill`
-
-`Thumbnail`, `Pad`, and `Fill` require a width to generate re-sampled images.
-Height can also be specified for these, but will default to the width.
-If no type is specified it will default to `Fill`.
+###### Image Transformation
+To cut down on 500 errors caused by trying to resize images when visiting a page images can be transformed by salsify.
+When an image transformation is update in the config it will also re-download the image with the new transformations.
```
Comment (muskie9) [inline README.md:289]: "updated" vs "update"?

File: src/TypeHandler/Asset/AssetHandler.php [hunk 2]
```diff
@@ -95,21 +97,53 @@ protected function findOrCreateFile($id, $class = File::class)
...
+     * @return bool
+     */
+    private function isTransformOutOfDate($file, $class, $transformation)
+    {
+        if (!$file instanceof Image) {
+            return false;
+        }
+
+        /** @var Image|ImageDataExtension $file */
+        if ($file->Transformation == $transformation) {
```
Comment (muskie9) [inline src/TypeHandler/Asset/AssetHandler.php:143]: Can we bring this down to `return $file->Transformation != $transformation;`?

**Review by muskie9 — CHANGES_REQUESTED** [review]
Looks great! One typo in the readme, and one suggestion for `isTransformOutOfDate()`. The latter just cuts down on lines and doesn't necessarily have to be implemented.

File: src/TypeHandler/Asset/AssetHandler.php (hunk 2 above)
Comment (mak001) [reply src/TypeHandler/Asset/AssetHandler.php:143]: possibly, but normal files don't have the transformation field, just images.

Comment (muskie9) [reply src/TypeHandler/Asset/AssetHandler.php:143]: Yup, this is just for this transformation logic at the tail end. We'd keep the instance check ,but bring this if block and subsequent returns down to the ternary. I don't think it's necessary required, just noting it.

**Pull request merged**